lib/base/RequestClient.js

```diff
@@ -45,13 +47,17 @@ RequestClient.prototype.request = function(opts) {
     headers.Authorization = 'Basic ' + b64Auth;
   }
 
+  if (opts.forever) {
+    var keepAliveAgent = new Agent({keepAlive: true});
```

Comment:
I don't like this scoping. Prefer something like: `agent: opts.forever === true ? new Agent({keepAlive: true}) : undefined `